src/controllers/users.controller.ts

```diff
@@ -0,0 +1,71 @@
+import {ValidatedRequest} from 'express-joi-validation';
+import * as userValidation from '../validation/validation-schemas';
+import UserService from '../services/user.service';
+import {Request, Response} from 'express';
+import {RESPONSE_MESSAGES} from '../config/messages';
+
+export class UsersController {
+  static async getSuggestedUsers(
+    req: ValidatedRequest<userValidation.UserRequestBodySchema>,
+    res: Response
+  ) {
+    const loginSubstring = req.query.loginSubstring || '';
+    const limit = req.query.limit ? +req.query.limit : undefined;
+    const suggestedUsers = await UserService.getSuggestedUsers(
+      loginSubstring,
+      limit
+    );
+    res.send(suggestedUsers);
+  }
+
+  static async getUser(req: Request, res: Response) {
+    const user = await UserService.getUser(req.params.id);
+    if (user) {
+      res.send(user);
+    } else {
+      res.status(404).send(RESPONSE_MESSAGES.USER_NOT_FOUND);
+    }
+  }
+
+  static async createUser(
+    req: ValidatedRequest<userValidation.UserRequestBodySchema>,
+    res: Response
+  ) {
+    const {age, login, password} = req.body;
+    const newUser = await UserService.createUser(age, login, password);
+    res.send(newUser);
+  }
+
+  static async editUser(
+    req: ValidatedRequest<userValidation.UserRequestBodySchema>,
+    res: Response
+  ) {
+    const {age, login, password} = req.body;
+    const existedUser = await UserService.getUser(req.params.id);
+    if (!existedUser) {
+      res.status(404).send(RESPONSE_MESSAGES.USER_NOT_FOUND);
+    } else {
+      const [, [updatedUser]] = await UserService.updateUser(
```

Comment:
I would return make updateUser method return user always. And if no user - > throw exception (all this logic inside updateUser service method()

Reply:
done